### PR TITLE
Version conflict exception message enhancement

### LINKED ISSUE #21278: Version conflict exception message enhancement
I am opening this issue to request the enhancement of the message when a document is not found. In that situation one gets:

https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/index/VersionType.java#L106

```
 "current version [" + currentVersion + "] is different than the one provided [" + expectedVersion + "]"
```

Being Current Version `-1`.

Maybe we can change the message when we have a document not found. Such as "Document not found". Which will be more intuitive than current version -1?

### PR DESCRIPTION
**Pull request by vladimirdolzhenko**
Report doc is not found rather on PUT ?version=X rather current version [-1] is different than the one provided

Closes #21278

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md)?
- If submitting code, have you built your formula locally prior to submission with `gradle check`?
- If submitting code, is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed.
- If submitting code, have you checked that your submission is for an [OS that we support](https://www.elastic.co/support/matrix#show_os)?
- If you are submitting this code for a class then read our [policy](https://github.com/elastic/elasticsearch/blob/master/CONTRIBUTING.md#contributing-as-part-of-a-class) for that.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/VersionType.java [hunk 1]
```diff
@@ -38,6 +38,9 @@ public String explainConflictForWrites(long currentVersion, long expectedVersion
             if (expectedVersion == Versions.MATCH_DELETED) {
                 return "document already exists (current version [" + currentVersion + "])";
             }
+            if (currentVersion == Versions.NOT_FOUND) {
+                return "document does not exist";
```
Comment (bleskes) [inline server/src/main/java/org/elasticsearch/index/VersionType.java]: can we say something about the version we expected to find?

File: server/src/main/java/org/elasticsearch/index/VersionType.java [hunk 2]
```diff
@@ -48,6 +51,9 @@ public boolean isVersionConflictForReads(long currentVersion, long expectedVersi
 
         @Override
         public String explainConflictForReads(long currentVersion, long expectedVersion) {
+            if (currentVersion == Versions.NOT_FOUND) {
+                return "document does not exist";
```
Comment (bleskes) [inline server/src/main/java/org/elasticsearch/index/VersionType.java]: same comment

File: server/src/main/java/org/elasticsearch/index/VersionType.java [hunk 3]
```diff
@@ -123,6 +129,9 @@ public boolean isVersionConflictForReads(long currentVersion, long expectedVersi
...
```
Comment (bleskes) [inline server/src/main/java/org/elasticsearch/index/VersionType.java]: same

File: server/src/main/java/org/elasticsearch/index/VersionType.java [hunk 4]
```diff
@@ -178,6 +187,9 @@ public boolean isVersionConflictForReads(long currentVersion, long expectedVersi
...
```
Comment (bleskes) [inline server/src/main/java/org/elasticsearch/index/VersionType.java]: same

**Review by bleskes — CHANGES_REQUESTED** [review]
Looks great. I left a minor ask.

**Comment by vladimirdolzhenko** [comment]
@bleskes amendments are done

**Review by bleskes — APPROVED** [review]
LGTM

**Pull request merged**